From 3126879fd777b21a8e6dff73251ceea821bd300f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 19 Jan 2022 09:47:49 +0200
Subject: [PATCH] [SW-58071] gaudi3: add regmap and regmask files

This patch adds regmap and regmask files for gaudi3 which are
used currently in common code.

Change-Id: Ib0bf8e9c5cc44cab88aa4b8b332fb044f4bb7134
---

diff --git a/arch/gaudi3/arc/regmap.h b/arch/gaudi3/arc/regmap.h
new file mode 100644
index 0000000..1ad4b79
--- /dev/null
+++ b/arch/gaudi3/arc/regmap.h
@@ -0,0 +1,75 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GAUDI3_REGMAP_H_
+#define _GAUDI3_REGMAP_H_
+
+#define mmI2C_S_IC_CON			mmD0_PSOC_I2C_S_IC_CON
+#define mmI2C_S_IC_SAR			mmD0_PSOC_I2C_S_IC_SAR
+#define mmI2C_S_IC_INTR_MASK		mmD0_PSOC_I2C_S_IC_INTR_MASK
+#define mmI2C_S_IC_RX_TL		mmD0_PSOC_I2C_S_IC_RX_TL
+#define mmI2C_S_IC_TX_TL		mmD0_PSOC_I2C_S_IC_TX_TL
+#define mmI2C_S_IC_ENABLE		mmD0_PSOC_I2C_S_IC_ENABLE
+
+#define mmQSPI_SER			mmD0_PSOC_QSPI_SER
+#define mmQSPI_BAUDR			mmD0_PSOC_QSPI_BAUDR
+#define mmQSPI_DR0			mmD0_PSOC_QSPI_DR0
+
+#define mmEFUSE_READ_CTRL		mmD0_PSOC_EFUSE_RD
+#define mmEFUSE_READ_ADDRESS		mmD0_PSOC_EFUSE_READ_ADDRESS
+#define mmEFUSE_DATA_0			mmD0_PSOC_EFUSE_DATA_0
+
+#define mmPRSTN				mmD0_PSOC_GLOBAL_CONF_PRSTN
+#define mmCOLD_RST_FLOPS_2		mmD0_PSOC_GLOBAL_CONF_COLD_RST_FLOPS_2
+#define mmPCIE_PHY_RST_N		mmD0_PSOC_GLOBAL_CONF_PCIE_PHY_RST_N
+#define mmBOOT_STRAP_PINS_L		mmD0_PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_L
+#define mmPRSTN_MASK			mmD0_PSOC_GLOBAL_CONF_PRSTN_MASK
+#define mmSPI_DMA_AWUSER		mmD0_PARC_GLOBAL_CONF_SPI_DMA_AWUSER
+#define mmSPI_DMA_CTRL			mmD0_PARC_GLOBAL_CONF_SPI_DMA_CTRL
+#define mmSPI_DMA_STATUS		mmD0_PARC_GLOBAL_CONF_SPI_DMA_STATUS
+#define mmSPI_DMA_DST_ADDR_L		mmD0_PARC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L
+#define mmSPI_DMA_DST_ADDR_H		mmD0_PARC_GLOBAL_CONF_SPI_DMA_DST_ADDR_H
+#define mmMEM_CPY_CTRL			mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL
+#define mmMEM_CPY_STATUS		mmD0_PSOC_GLOBAL_CONF_MEM_CPY_STATUS
+#define mmMEM_CPY_DEST_ADDR_H		mmD0_PSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_H
+#define mmMEM_CPY_DEST_ADDR_L		mmD0_PSOC_GLOBAL_CONF_MEM_CPY_DEST_ADDR_L
+#define mmMEM_CPY_CTRL2			mmD0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2
+#define mmMEM_CPY_AXUSER		mmD0_PSOC_GLOBAL_CONF_MEM_CPY_AXUSER
+#define mmSPI_DIRECT_WR_RD_CTRL		mmD0_PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL
+
+#define mmGPIO0_DATA			mmD0_PSOC_GPIO0_DATA
+
+#define mmSECURITY_MIN_FW_VERSION	mmD0_PSOC_SECURITY_MIN_FW_VERSION
+#define mmSECURITY_PRI_IMG_STATUS	mmD0_PSOC_SECURITY_PRI_IMG_STATUS
+#define mmSECURITY_SEC_IMG_STATUS	mmD0_PSOC_SECURITY_SEC_IMG_STATUS
+#define mmSECURITY_PRI_IMG_FW_VER	mmD0_PSOC_SECURITY_PRI_IMG_FW_VER
+#define mmSECURITY_SEC_IMG_FW_VER	mmD0_PSOC_SECURITY_SEC_IMG_FW_VER
+#define mmSECURITY_STATUS		mmD0_PSOC_SECURITY_STATUS
+#define mmSECURITY_KEY_REVOKE		mmD0_PSOC_SECURITY_KEY_REVOKE
+#define mmSW_STATUS_0			mmD0_PSOC_SECURITY_SW_STATUS_0
+
+#define mmJT_MSTR_IF_CORE_HBW_AXPROT_OVD	mmD0_JT_MSTR_IF_AXPROT_HBW_ARPROT_0_OVRD
+
+#define mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVD	mmD0_SMI_MSTR_IF_AXPROT_HBW_ARPROT_0_OVRD
+
+#define mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVD	mmD0_I2C_S_MSTR_IF_AXPROT_HBW_ARPROT_0_OVRD
+
+#define mmCPU_HOT_RST			mmD0_PCIE_CORE_HOT_RST
+#define mmPCIE_RSP_OK			mmD0_PCIE_WRAP_PCIE_RSP_OK
+
+#define mmLINK_CAPABILITIES_REG			mmD0_PCIE_DBI_SIG_LINK_CAPABILITIES_REG
+#define mmLINK_CONTROL2_LINK_STATUS2_REG	mmD0_PCIE_DBI_SIG_LINK_CONTROL2_LINK_STATUS2_REG
+
+#define mmLTSSM_EN			mmD0_PCIE_AUX_LTSSM_EN
+#define mmAPP_SRIS_MODE			mmD0_PCIE_AUX_APP_SRIS_MODE
+
+#define mmPHY_EXT_CTRL_SEL		mmD0_PCIE_PHY_PHY_EXT_CTRL_SEL
+#define mmPIPE_RX_SRIS_MODE_EN		mmD0_PCIE_PHY_PIPE_RX_SRIS_MODE_EN
+#define mmEXT_MPLLA_SSC_PEAK		mmD0_PCIE_PHY_EXT_MPLLA_SSC_PEAK
+#define mmEXT_MPLLA_SSC_STEP_SIZE	mmD0_PCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE
+#define mmEXT_MPLLB_SSC_PEAK		mmD0_PCIE_PHY_EXT_MPLLB_SSC_PEAK
+#define mmEXT_MPLLB_SSC_STEP_SIZE	mmD0_PCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE
+
+#endif /* _GAUDI3_REGMAP_H_ */
diff --git a/arch/gaudi3/arc/regmask.h b/arch/gaudi3/arc/regmask.h
new file mode 100644
index 0000000..041ef0c
--- /dev/null
+++ b/arch/gaudi3/arc/regmask.h
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ */
+
+#ifndef _GAUDI3_REGMASK_H_
+#define _GAUDI3_REGMASK_H_
+
+#define SPI_DMA_CTRL_ADDR_SHIFT			D0_PARC_GLOBAL_CONF_SPI_DMA_CTRL_ADDR_SHIFT_S
+#define SPI_DMA_STATUS_ERROR_MASK		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_M
+#define SPI_DMA_STATUS_ERROR_SHIFT		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_ERROR_S
+#define SPI_DMA_STATUS_DONE_MASK		D0_PARC_GLOBAL_CONF_SPI_DMA_STATUS_DONE_M
+#define SPI_DMA_CTRL_FLD_SIZE_SHIFT		D0_PARC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_S
+#define SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_MASK	\
+					D0_PSOC_GLOBAL_CONF_SPI_DIRECT_WR_RD_CTRL_BYTE_SWAP_M
+#define PRSTN_MASK_IND_MASK			D0_PSOC_GLOBAL_CONF_PRSTN_MASK_IND_M
+#define MEM_CPY_CTRL2_USE_CONST_SHIFT		D0_PSOC_GLOBAL_CONF_MEM_CPY_CTRL2_USE_CONST_S
+
+#define SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_SHIFT	\
+					D0_PSOC_SECURITY_PRI_IMG_STATUS_FW_VER_LOWER_S
+#define SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_SHIFT	\
+					D0_PSOC_SECURITY_PRI_IMG_STATUS_AUTHENTICATE_FAIL_S
+#define SECURITY_PRI_IMG_STATUS_CRC_FAIL_SHIFT	D0_PSOC_SECURITY_PRI_IMG_STATUS_CRC_FAIL_S
+#define SECURITY_STATUS_IMG_VALIDATED_SHIFT	D0_PSOC_SECURITY_STATUS_IMG_VALIDATED_S
+#define SECURITY_STATUS_FW_UPGRADE_SHIFT	D0_PSOC_SECURITY_STATUS_FW_UPGRADE_S
+#define SECURITY_STATUS_KEY_REVOKED_SHIFT	D0_PSOC_SECURITY_STATUS_KEY_REVOKED_S
+#define SECURITY_STATUS_KEY_REVOKED_IND_SHIFT	D0_PSOC_SECURITY_STATUS_KEY_REVOKED_IND_S
+#define SECURITY_STATUS_MIN_FW_MODIFIED_SHIFT	D0_PSOC_SECURITY_STATUS_MIN_FW_MODIFIED_S
+#define SECURITY_STATUS_FW_VER_INC_SHIFT	D0_PSOC_SECURITY_STATUS_FW_VER_INC_S
+#define SECURITY_STATUS_SPI_IMG_MODIFIES_SHIFT	D0_PSOC_SECURITY_STATUS_SPI_IMG_MODIFIES_S
+#define SECURITY_STATUS_FW_AUTHENTICAION_FAIL_SHIFT	\
+					D0_PSOC_SECURITY_STATUS_FW_AUTHENTICAION_FAIL_S
+
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_MASK	\
+				D0_PCIE_DBI_SIG_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_M
+#define LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_SHIFT	\
+				D0_PCIE_DBI_SIG_LINK_CAPABILITIES_REG_PCIE_CAP_MAX_LINK_SPEED_S
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_MASK	\
+			D0_PCIE_DBI_SIG_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_M
+#define LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_SHIFT	\
+			D0_PCIE_DBI_SIG_LINK_CONTROL2_LINK_STATUS2_REG_PCIE_CAP_TARGET_LINK_SPEED_S
+
+#define LTSSM_EN_IND_MASK			D0_PCIE_AUX_LTSSM_EN_IND_M
+
+#define CPU_HOT_RST_CTRL_MASK			D0_PCIE_CORE_HOT_RST_INTR_M
+
+#endif /* _GAUDI3_REGMASK_H_ */
